File: wmoecke-cs50-problems-2023-x-sentimental-mario-more/mario.py
from cs50 import get_int


def main():
    n = 0

    # Get user input for height
    while n < 1 or n > 8:
        n = get_int("Height: ")

    # Draw the sprite per user input
    for row in range(n - 1, -1, -1):
        draw_tiles(' ', row)
        draw_tiles('#', n - row)
        draw_tiles(' ', 2)
        draw_tiles('#', n - row)
        print()


def draw_tiles(tile, quantity):
    """Function to draw the tiles in a row according to quantity"""
    for i in range(quantity):
        print(tile, end='')


if __name__ == "__main__":
    main()

File: wmoecke-cs50-problems-2023-x-mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void draw_tiles(char sprite, int n);

int main(void)
{
    int n = 0;

    // Get user input for height
    while (n < 1 || n > 8)
    {
        n = get_int("Height: ");
    }

    // Draw the sprite per user input
    for (int row = n - 1; row >= 0; row--)
    {
        draw_tiles(' ', row);
        draw_tiles('#', n - row);
        draw_tiles(' ', 2);
        draw_tiles('#', n - row);
        printf("\n");
    }
}

// Function to draw the tiles in a row according to quantity
void draw_tiles(char tile, int quantity)
{
    for (int i = 0; i < quantity; i++)
    {
        printf("%c", tile);
    }
}